### ISSUE TITLE
Ignore <records>false</records> for UDTRecords

### ISSUE THREAD

**Issue by Mosrod** [jOOQ/jOOQ#10529]
### Expected behavior and actual behavior:
Well, I want there to not be any errors :)

Here is my schema structure:
![image](https://user-images.githubusercontent.com/18247709/91233608-993a3880-e6ff-11ea-8a17-837422bb2760.png)

And here is the code to create the Address type in Postgres:
drop type if exists limehrm.address;
```
create type limehrm.address as (
    street_name text,
    apartment_or_unit_number text,
    city text,
    state text,
    zip text,
    country text
);

```
The enums work fine, but I get an error in the udt/Address.java generated class. 
![image](https://user-images.githubusercontent.com/18247709/91233492-4d878f00-e6ff-11ea-81d9-54bc0e1c0f1b.png)

Here is a sample of the Address java class, which is trying to import a class named AddressRecord which doesn't exist:
![image](https://user-images.githubusercontent.com/18247709/91233562-79a31000-e6ff-11ea-898b-94dfc1f93026.png)

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

Generate Schema in Postgres file:

```
drop schema if exists limehrm cascade;

create schema limehrm;

/* create custom types */
drop type if exists limehrm.address;
create type limehrm.address as (
    street_name text,
    apartment_or_unit_number text,
    city text,
    state text,
    zip text,
    country text
);

drop type if exists limehrm.sex;
create type limehrm.sex as enum ('male', 'female', 'other');

drop type if exists limehrm.marital_status;
create type limehrm.marital_status as enum ('single', 'married');

drop type if exists limehrm.job_status;
create type limehrm.job_status as enum ('ongoing', 'terminated', 'completed');

drop table if exists workers;

create table limehrm.workers (
    id int,
    first_name text,
    last_name text,
    personal_email text,
    home_phone text,
    home_address limehrm.address,
    sex limehrm.sex,
    marital_status limehrm.marital_status,
    position_name text,
    team_name text,
    salary bool,
    department text,
    birth_date date,
    start_date date,
    end_date date,
    job_status limehrm.job_status,
    manager_email text
);
```
jOOQ generator configuration:

```
jooq {
    configurations {
        create("main") {
            generateSchemaSourceOnCompilation.set(true)
            
            jooqConfiguration.apply { 
                logging = org.jooq.meta.jaxb.Logging.WARN
                jdbc.apply {
                    driver = "org.postgresql.Driver"
                    url = "jdbc:postgresql://localhost:5432/postgres"
                    user = "postgres"
                    password = "postgres"
//                    properties.add(Property().withKey("ssl").withValue("true"))
                }

                generator.apply { 
                    name = "org.jooq.codegen.DefaultGenerator"
                    database.apply { 
                        name = "org.jooq.meta.postgres.PostgresDatabase"
                        inputSchema = "limehrm"

                        forcedTypes.addAll(arrayOf(
                            ForcedType()
                                    .withName("varchar")
                                    .withIncludeExpression(".*")
                                    .withIncludeTypes("JSONB?"),
                            ForcedType()
                                    .withName("varchar")
                                    .withIncludeExpression(".*")
                                    .withIncludeTypes("INET")
                        ).toList())
                    }
                    
                    generate.apply { 
                        isDeprecated = false
                        isRecords = false
                        isImmutablePojos = true
                        isFluentSetters = true
                    }
                    
                    target.apply { 
                        packageName = "limehrm"
                        directory = "build/generated-src/jooq/main"
                    }
                    
                    strategy.name = "org.jooq.codegen.DefaultGeneratorStrategy"
                }
            }
        }
    }
}
```

### Versions:

- jOOQ: 3.13.4
- Java: 14
- Database (include vendor): Postgres 12.4
- OS: Windows
- JDBC Driver (include name if inofficial driver): org.postgresql.Driver

**Comment by Mosrod** [comment]
Works perfectly with the Java generator, seems to be an issue with the Gradle generator plugin @etiennestuder

**Comment by lukaseder** [comment]
Thanks a lot for your report.

The reason for this seems to be your configuration to avoid records:

```
isRecords = false
```

Remove that line, and it should work. Of course, with your configuration, we should still try to produce valid Java code. I wonder if it's possible with UDTs, you kinda need the generated `UDTRecord` to work with them (unlike the `TableRecord`). So, there are two possible fixes here:

- Try to get this to compile, producing `extends UDTImpl<UDTRecord<?>>`
- Ignore the flag for UDTs

> Works perfectly with the Java generator, seems to be an issue with the Gradle generator plugin @etiennestuder

You probably meant "maven generator", not "Java generator"? Can you confirm that when using Maven, you didn't turn off `<records>false</records>`? I don't think the Gradle plugin can do anything wrong here. It just takes care of configuring jOOQ's `GenerationTool` and makes use of a few Gradle features to leverage incremental compilation. It doesn't touch the actual code generation in a way that compilation errors could be traced back to it.

**Comment by etiennestuder** [comment]
> It doesn't touch the actual code generation in a way that compilation errors could be traced back to it.

Yes, this cannot be an issue with the plugin, since in the end, it passes a jooq config xml file to the GenerationTool. I will close this issue.

**Comment by lukaseder** [comment]
> Oh this is a jOOQ issue - nothing to close for me :-)

Beat me to it :)

**Comment by Mosrod** [comment]
Ah, sorry about that; didn't realize how the plugin worked. I'm also not sure why I set records to false, must have been a typo.

It does work on the Gradle plugin when setting records to true and thank you for spotting this error :)

I was experimenting with generating the code programmatically using [this code](https://stackoverflow.com/a/63603773/9150279) and isRecords is defaulted to True when using this approach.

Again, thank you for your quick response and solving my issue :-)

**Comment by lukaseder** [comment]
You're welcome, @Mosrod, and thanks again for reporting this.

**Comment by lukaseder** [comment]
Running into this now, myself. Looking into a fix for jOOQ 3.15

**Comment by lukaseder** [comment]
`UDTRecordImpl` cannot be used here. We need a new final subclass `DefaultUDTRecord` and make some adaptations to jOOQ's internals - so this is not only a code generation issue.

**Comment by lukaseder** [comment]
With the new API, I prefer not to backport the fix to 3.14. Fixed in 3.15 only

**Comment by lukaseder** [comment]
This approach just causes headaches when nesting UDTs and other data types. Perhaps it's better to just ignore `<records>false</records>` for UDTs and always generate `UDTRecord` types

**Comment by lukaseder** [comment]
With that, it's a trivial fix for 3.15.0 and 3.14.4 (#10925)
